api/types/presence.go

```diff
@@ -70,6 +70,9 @@ func (s *KeepAlive) CheckAndSetDefaults() error {
 	if s.Namespace == "" {
 		s.Namespace = defaults.Namespace
 	}
+	if s.Namespace != defaults.Namespace {
+		return trace.BadParameter("namespace %q invalid, custom namespaces are deprecated", s.Namespace)
+	}
```

Comment:
nit: in this and the other error messages like it, we probably want to explicitly tell the user how to fix the problem.  Ex:

```suggestion
	if s.Namespace != defaults.Namespace {
		return trace.BadParameter("namespace %q invalid, custom namespaces are deprecated, the namespace field should be omitted or set to %q", s.Namespace, defaults.Namespace)
	}
```

Reply:
Done using a validation helper. I'm not even sure users can set the field (in most resources the json tag is set to "-"), but code definitely can (and some resources seem to use protojson which bypasses the gogojson tag).